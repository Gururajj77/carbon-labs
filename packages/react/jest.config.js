module.exports = {
  testEnvironment: 'jsdom',
  preset: 'ts-jest',
  moduleNameMapper: {
    '\\.(css|less|scss|sass)$': '<rootDir>/__mocks__/styleMock.js',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
  transform: {
    '^.+\\.(ts|tsx)$': 'ts-jest',
    '^.+\\.(js|jsx)$': ['babel-jest', { configFile: './.babelrc' }],
  },
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],

  // Test file patterns to match
  testMatch: ['**/__tests__/**/*.[jt]s?(x)', '**/?(*.)+(spec|test).[jt]s?(x)'],
};
