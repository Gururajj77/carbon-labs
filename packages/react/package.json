{
  "name": "my-app",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "storybook": "storybook dev -p 3000",
    "build-storybook": "storybook build"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "dependencies": {
    "@carbon/react": "^1.71.1",
    "@carbon/styles": "^1.70.0",
    "@testing-library/jest-dom": "^5.17.0",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.5.2",
    "@types/node": "^16.18.121",
    "@types/react": "^18.3.12",
    "@types/react-dom": "^18.3.1",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "typescript": "^5.2.2"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.26.0",
    "@babel/preset-react": "^7.26.3",
    "@chromatic-com/storybook": "^3.2.2",
    "@storybook/addon-docs": "^8.4.6",
    "@storybook/addon-essentials": "^8.4.6",
    "@storybook/addon-interactions": "^8.4.6",
    "@storybook/addon-onboarding": "^8.4.6",
    "@storybook/addon-storysource": "^8.4.6",
    "@storybook/addon-webpack5-compiler-babel": "^3.0.3",
    "@storybook/blocks": "^8.4.6",
    "@storybook/react": "^8.4.6",
    "@storybook/react-webpack5": "^8.4.6",
    "@storybook/test": "^8.4.6",
    "@types/babel__preset-env": "^7",
    "@types/prop-types": "^15",
    "css-loader": "^7.1.2",
    "eslint-plugin-storybook": "^0.11.1",
    "mini-css-extract-plugin": "^2.9.2",
    "postcss-loader": "^8.1.1",
    "prop-types": "^15.8.1",
    "remark-gfm": "^4.0.0",
    "sass-loader": "^16.0.4",
    "storybook": "^8.4.6",
    "storybook-addon-accessibility-checker": "^3.1.61-rc.3",
    "style-loader": "^4.0.0",
    "typescript-config-carbon": "^0.3.0",
    "webpack": "^5.97.0"
  }
}
