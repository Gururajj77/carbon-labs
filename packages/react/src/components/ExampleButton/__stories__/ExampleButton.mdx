import { ArgTypes, Canvas, Meta } from '@storybook/blocks';
import * as ExampleButtonStories from './ExampleButton.stories';

<Meta isTemplate />

# ExampleButton

[Source code](https://github.com/carbon-design-system/carbon/tree/main/packages/react/src/components/Accordion)
&nbsp;|&nbsp;
[Usage guidelines](https://www.carbondesignsystem.com/components/accordion/usage)
&nbsp;|&nbsp;
[Accessibility](https://www.carbondesignsystem.com/components/accordion/accessibility)

{/* <!-- START doctoc generated TOC please keep comment here to allow auto update --> */}
{/* <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE --> */}

## Table of Contents

- [Overview](#overview)
- [Skeleton state](#skeleton-state)
- [Component API](#component-api)
- [References](#references)
- [Feedback](#feedback)

{/* <!-- END doctoc generated TOC please keep comment here to allow auto update --> */}

## Overview

You can build an accordion using a combination of the `Accordion` and
`AccordionItem` components. The `Accordion` components accepts a list of
`AccordionItem` components as `children` and each `AccordionItem` is responsible
for displaying the accordion's heading and panel content.

You can configure the accordion item's heading using the `title` prop.
Everything you pass in as a child of `AccordionItem` will be rendered in the
accordion's panel.

<Canvas of={ExampleButtonStories.Default} />
