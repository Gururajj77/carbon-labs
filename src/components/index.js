export { default as C4AIExtendedButton } from "./extended-button/extended-button";
export { default as C4AITestInput } from "./test-input/test-input";
