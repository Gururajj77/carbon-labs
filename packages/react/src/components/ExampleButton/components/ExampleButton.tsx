import React, { PropsWithChildren } from 'react';
import { Button, ButtonProps } from '@carbon/react';
import './button.scss';

/** Primary UI component for user interaction */

type ExampleButtonProps = PropsWithChildren<ButtonProps<any>>;
export const ExampleButton = ({ children = 'Button', ...rest }: ExampleButtonProps ) => {
  return <Button {...rest}>{children}</Button>;
};