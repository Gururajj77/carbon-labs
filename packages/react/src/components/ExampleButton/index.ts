import { ExampleButton } from './components/ExampleButton';

export { ExampleButton };
