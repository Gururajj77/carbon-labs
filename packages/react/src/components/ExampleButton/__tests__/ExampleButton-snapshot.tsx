// Component.test.tsx
import { render } from '@testing-library/react';
import {ExampleButton} from '../components/ExampleButton';
import React from 'react';
describe('YourComponent', () => {
  it('should match snapshot', () => {
    const { container } = render(<ExampleButton />);
    expect(container).toMatchSnapshot();
  });
});