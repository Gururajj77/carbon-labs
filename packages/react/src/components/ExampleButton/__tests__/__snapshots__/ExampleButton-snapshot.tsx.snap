// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourComponent should match snapshot 1`] = `
<div>
  <button
    class="cds--btn cds--btn--primary"
    type="button"
  >
    Button
  </button>
</div>
`;
